### PR TITLE
docs: refine examples & add missing DNS overview

### PR DESCRIPTION
**Pull request by stephenplusplus**
While working on the Compute Engine PR, I caught a few doc-related issues in our code.

Here's a preview: https://stephenplusplus.github.io/gcloud-node/#/docs/

### CODE REVIEW

File: docs/site/components/docs/docs.html [hunk 1]
```diff
@@ -44,7 +44,7 @@ <h3 class="sub-heading">
     </article>
     <hr>
 
-    <article ng-repeat="service in ['bigquery', 'datastore', 'pubsub', 'search', 'storage']"
+    <article ng-repeat="service in ['bigquery', 'datastore', 'dns', 'pubsub', 'search', 'storage']"
```
Comment (stephenplusplus) [inline docs/site/components/docs/docs.html]: I forgot the DNS overview -- created a file for it (see dns-overview.html)

File: docs/site/components/docs/pubsub-overview.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <h3>Pub/Sub Overview</h3>
 <p class="notice">
-  Google Cloud Pub/Sub is in Alpha status. As a result, it might be changed in backward-incompatible ways and is not recommended for production use. It is not subject to any SLA or deprecation policy.
+  <strong>This is a Beta release of Google Cloud Pub/Sub.</strong> This feature is not covered by any SLA or deprecation policy and may be subject to backward-incompatible changes.
```
Comment (stephenplusplus) [inline docs/site/components/docs/pubsub-overview.html]: Just changed the style and wording for consistency.

File: docs/site/components/docs/search-overview.html [hunk 3]
```diff
@@ -1,4 +1,7 @@
 <h3>Search Overview</h3>
+<p class="notice">
+  <strong>This is an Alpha release of Google Cloud Search.</strong> This feature is not covered by any SLA or deprecation policy and may be subject to backward-incompatible changes.
+</p>
```
Comment (stephenplusplus) [inline docs/site/components/docs/search-overview.html]: Previously forgot to include this warning.

File: lib/index.js [hunk 4]
```diff
@@ -96,30 +96,30 @@ var util = require('./common/util.js');
  *
  * @example
  * var gcloud = require('gcloud')({
- *   keyFilename: '/path/to/keyfile.json',
- *   projectId: 'my-project'
+ *   projectId: 'grape-spaceship-123',
+ *   keyFilename: '/path/to/keyfile.json'
```
Comment (stephenplusplus) [inline lib/index.js]: All examples synced to use projectId (grape-spaceship-123) -> keyFilename. This is consistent across our docs.

File: lib/index.js [hunk 5]
```diff
@@ -96,30 +96,30 @@ var util = require('./common/util.js');
...
- * @example
- * var gcloud = require('gcloud')({
- *   keyFilename: '/path/to/keyfile.json',
- *   projectId: 'my-project',
- *   maxRetries: 5 // retry rate limited requests 5 times before giving up
- * });
+ * //-
+ * // `gcs` and `otherGcs` will use their respective credentials for all future
+ * // API requests.
+ * //-
```
Comment (stephenplusplus) [inline lib/index.js]: Hopefully this example is more clear about overriding authentication details.

File: lib/index.js [hunk 6]
```diff
@@ -180,24 +180,42 @@ gcloud.bigquery = BigQuery;
...
+ *
+ * @return {module:dns}
+ *
+ * @example
+ * var gcloud = require('gcloud');
+ * var dns = gcloud.dns({
+ *   projectId: 'grape-spaceship-123',
+ *   keyFilename: '/path/to/keyfile.json'
+ * });
+ */
```
Comment (stephenplusplus) [inline lib/index.js]: Forgot to document here!

File: lib/index.js [hunk 7]
```diff
@@ -180,24 +180,42 @@ gcloud.bigquery = BigQuery;
...
  * Platform.
  *
- * Note: This is a *Beta* release of Google Cloud Pub/Sub. This feature is not
- * covered by any SLA or deprecation policy and may be subject to backward-
- * incompatible changes.
+ * <p class="notice">
+ *   **This is a Beta release of Google Cloud Pub/Sub.** This feature is not
+ *   covered by any SLA or deprecation policy and may be subject to backward-
+ *   incompatible changes.
+ * </p>
```
Comment (stephenplusplus) [inline lib/index.js]: Again, consistent messaging / styling.

File: lib/index.js [hunk 8]
```diff
@@ -206,25 +224,23 @@ gcloud.dns = DNS;
...
  * var pubsub = gcloud.pubsub({
- *   projectId: 'project-id',
+ *   projectId: 'grape-spaceship-123',
  *   keyFilename: '/path/to/keyfile.json'
  * });
  */
-gcloud.pubsub = function(config) {
-  return new PubSub(config);
-};
+gcloud.pubsub = PubSub;
```
Comment (stephenplusplus) [inline lib/index.js]: Awkwardly, `pubsub` and `search` were doing more work than necessary. I adopted the system we use for our other APIs, and we now do auto-instantiation in the constructor, e.g.:

```js
function PubSub(options) {
  if (!(this instanceof PubSub)) {
    return new PubSub(options);
  }
  // ...
}
```

**Comment by callmehiphop** [comment]
Nice catches!

**Pull request merged**